vibe_code_apps_22/ml/weather_forecast_48h.py

```
@@ -0,0 +1,135 @@
+            "appid": api_key,
+        }
+    )
+    url = f"https://api.openweathermap.org/data/3.0/onecall?{q}"
+    with urllib.request.urlopen(url, timeout=timeout_s) as resp:
+        doc = json.loads(resp.read().decode("utf-8"))
+    hours = doc.get("hourly") or []
+    oat, rh, ghi = [], [], []
+    for row in hours[:48]:
+        oat.append(float(row.get("temp", 20.0)))
+        rh.append(float(row.get("humidity", 50.0)))
+        # OWM clouds 0–100 → crude GHI proxy
```

Ruff (0.16.1)

[warning] Comment contains ambiguous `–` (EN DASH). Did you mean `-` (HYPHEN-MINUS)?

(RUF003)